libraries/swapfile_provider.rb

```diff
@@ -177,6 +178,28 @@ def persist
           ::File.open(fstab, 'w') { |f| f.write(contents.join('')) }
         end
       end
+
+      # According to Ubuntu and Redhat comunity document, A value of swappiness=10 is recommended
+      # Link: https://help.ubuntu.com/community/SwapFaq#What_is_swappiness_and_how_do_I_change_it.3F
+      # Link: https://access.redhat.com/documentation/en-US/Red_Hat_Enterprise_Linux/6/html/Performance_Tuning_Guide/s-memory-tunables.html
+      def swappiness
+        proc_command = 'echo 10 > /proc/sys/vm/swappiness'
+        Chef::Log.info("#{@new_resource} is setting swappiness value in '/proc/sys/vm/swappiness'")
+        shell_out!(proc_command)
+
+        sysctl = '/etc/sysctl.conf'
```

Comment:
Chef has a `Chef::Util::FileEdit` class that can be used to modify these types of partial updates. The current implementation is not idempotent or atomic.